src/Psalm/Internal/Analyzer/Statements/Expression/BinaryOp/NonDivArithmeticOpAnalyzer.php

```diff
@@ -387,12 +387,18 @@ private static function analyzeNonDivOperands(
             && $right_type_part instanceof TInt
             && $parent instanceof PhpParser\Node\Expr\PostInc
```

Comment:
And what about preincrement? Doesn't it need a similar change?

Reply:
I added pre inc and also pre/post dec. Thanks

Reply:
I'd suggest to add a test for that. Something simple, like asserting result of '123'++, ++'123' etc

Reply:
I did. This led to some more code to support literals.

I removed PreDec and PostDec though because they're handled totally differently in IncDecExpressionAnalyzer and I couldn't really figure why

Reply:
> I did. This led to some more code to support literals.

:+1: